src/adapter/graphql/src/mutations/flows_mut/flows_mut_utils.rs

```diff
@@ -128,6 +129,46 @@ pub(crate) async fn ensure_flow_preconditions(
             };
         }
         DatasetFlowType::HardCompaction => (),
+        DatasetFlowType::Reset => {
+            if let Some(flow_configuration) = flow_run_configuration
+                && let FlowRunConfiguration::Reset(reset_configuration) = flow_configuration
+            {
+                let dataset_repo = from_catalog::<dyn kamu_core::DatasetRepository>(ctx).unwrap();
+
+                let dataset = dataset_repo
+                    .get_dataset(&dataset_handle.as_local_ref())
+                    .await
+                    .int_err()?;
+                let new_head_block_result = dataset
+                    .as_metadata_chain()
+                    .get_block(&reset_configuration.new_head_hash)
+                    .await;
```

Comment:
Method became too large, split it on parts for each flow type.

Reply:
Almost all this logic moved to another levels